packages/plugins/plugin-appsflyer/src/AppsflyerPlugin.tsx

```diff
@@ -3,27 +3,77 @@ import {
   IdentifyEventType,
   PluginType,
   TrackEventType,
+  UpdateType,
+  SegmentAPISettings,
 } from '@segment/analytics-react-native';
-import appsFlyer, { InitSDKOptions } from 'react-native-appsflyer';
+import type { SegmentAppsflyerSettings } from './types';
+import appsFlyer from 'react-native-appsflyer';
 import identify from './methods/identify';
 import track from './methods/track';
 
 export class AppsflyerPlugin extends DestinationPlugin {
   type = PluginType.destination;
-  key = 'Appsflyer';
+  key = 'AppsFlyer';
 
-  constructor(opts: InitSDKOptions) {
-    super();
+  private settings: SegmentAppsflyerSettings | null = null;
+  private hasRegisteredInstallCallback: Boolean = false;
+  private hasRegisteredDeepLinkCallback: Boolean = false;
 
-    const defaultOpts = {
+  private hasInitialized: Boolean = false;
+
+  update(settings: SegmentAPISettings, _: UpdateType) {
+    let defaultOpts = {
       isDebug: true,
       timeToWaitForATTUserAuthorization: 60,
+      onInstallConversionDataListener: true,
     };
 
-    appsFlyer.initSdk({
-      ...defaultOpts,
-      ...opts,
-    });
+    const appsflyerSettings = settings.integrations[
+      this.key
+    ] as SegmentAppsflyerSettings;
+
+    if (!appsflyerSettings) {
+      return;
+    }
+    const clientConfig = this.analytics?.getConfig();
+
+    this.settings = appsflyerSettings;
+
+    if (
+      this.settings.trackAttributionData === true &&
+      this.hasRegisteredInstallCallback === false
+    ) {
+      this.conversionCallback();
+      this.hasRegisteredInstallCallback = true;
+    }
+
+    if (
+      clientConfig?.trackDeepLinks === true &&
+      this.hasRegisteredDeepLinkCallback === false
+    ) {
+      this.deepLinkCallback();
+      this.hasRegisteredDeepLinkCallback = true;
+    }
+
+    if (
+      this.hasInitialized === false &&
+      clientConfig?.trackDeepLinks === true
+    ) {
+      appsFlyer.initSdk({
+        devKey: this.settings.appsFlyerDevKey,
+        appId: this.settings.appleAppID,
+        onDeepLinkListener: true,
+        ...defaultOpts,
+      });
+      this.hasInitialized = true;
+    } else {
+      appsFlyer.initSdk({
+        devKey: this.settings.appsFlyerDevKey,
+        appId: this.settings.appleAppID,
+        ...defaultOpts,
+      });
+      this.hasInitialized = true;
+    }
```

Comment:
This `else` needs a condition since this will execute every time, as `this.hasInitialized = true` gets here.

The correct way would be to check again `else if (!this.hasInitialized)`.

However there's only one param that changes when `clientConfig.trackDeepLinks` is `true` so this simplifies it:

```
if (!this.hasInitialized) {
  appsFlyer.initSdk({
    devKey: this.settings.appsFlyerDevKey,
    appId: this.settings.appleAppID,
    onDeepLinkListener: clientConfig?.trackDeepLinks === true,
    ...defaultOpts,
  });
  this.hasInitialized = true;
}
```